src/uu/cp/src/cp.rs

```diff
@@ -1375,6 +1375,17 @@ fn handle_existing_dest(
     Ok(())
 }
 
+/// Decide whether the given path exists.
+fn file_or_link_exists(path: &Path) -> bool {
+    // Using `Path.exists()` or `Path.try_exists()` is not sufficient,
+    // because if `path` is a symbolic link and there are too many
+    // levels of symbolic link, then those methods will return false
+    // or an OS error.
+    std::fs::metadata(path)
```

Comment:
Is this not functionally the same as just using `symlink_metadata`? Because if the path is not a link, they are the same and if it is a link, then `symlink_metadata` should return `Ok` in a superset of the cases of `metadata`, right?

```rust
fn file_or_link_exists(path: &Path) -> bool {
    path.symlink_metadata().is_ok()
}
```

Reply:
I think you might be right. I'll update it.